File: php/features.php
<?php
declare(strict_types=1);

// FastlineVpn SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FastlineVpnFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FastlineVpnBaseFeature();
            case "test":
                return new FastlineVpnTestFeature();
            default:
                return new FastlineVpnBaseFeature();
        }
    }
}

File: py/features.py
# FastlineVpn SDK feature factory

from feature.base_feature import FastlineVpnBaseFeature
from feature.test_feature import FastlineVpnTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FastlineVpnBaseFeature(),
        "test": lambda: FastlineVpnTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
